stateproof/builder.go

```diff
@@ -165,11 +165,31 @@ func (spw *Worker) handleSig(sfa sigFromAddr, sender network.Peer) (network.Forw
 	spw.mu.Lock()
 	defer spw.mu.Unlock()
 
+	// The sig should be for a round which is a multiple of StateProofInterval
```

Comment:
I would suggest moving this code to be executed after
```
builderForRound, ok := spw.builders[sfa.Round]
	if !ok {
```
if the round is present we spend time on these checks.


+ could you export this code into a different function?